client/my-sites/themes/activation-modal.jsx

```diff
@@ -45,28 +47,51 @@ export class ActivationModal extends Component {
 		newThemeId: PropTypes.string,
 	};
 
+	constructor( props ) {
+		super( props );
+		const { isCurrentThemeAllowedOnSite } = props;
+
+		this.state = {
+			checkboxChecked: isCurrentThemeAllowedOnSite,
+		};
+	}
+
+	onCheckboxChange = ( isChecked ) => {
+		this.setState( { checkboxChecked: isChecked } );
+	};
+
 	closeModalHandler =
 		( action = 'dismiss' ) =>
 		() => {
-			const { newThemeId, siteId, source } = this.props;
+			const { newThemeId, activeTheme, siteId, source, isCurrentThemeAllowedOnSite } = this.props;
 			if ( 'activeTheme' === action ) {
 				this.props.acceptActivationModal( newThemeId );
+				const eventName = ! isCurrentThemeAllowedOnSite
+					? 'calypso_theme_switch_plan_warning_accepted'
+					: 'calypso_theme_autoloading_homepage_modal_activate_click';
```

Comment:
i think we should keep recording the `calypso_theme_autoloading_homepage_modal_activate_click` event regardless of whether the theme is allowed on the site, since we might be using that event for some metrics.

Maybe we can include an additional prop:

```js
recordTracksEvent( 'calypso_theme_autoloading_homepage_modal_activate_click', {
	theme: newThemeId,
	activeTheme: activeTheme.id,
	isActiveThemeAllowedOnSite: isCurrentThemeAllowedOnSite,
} );
```

Reply:
As explained on the PR description, I prefer to have them separate so that we can monitor each case individually.